modules/tensor_mechanics/include/materials/StressUpdateBase.h

```diff
@@ -36,10 +36,15 @@ class StressUpdateBase : public Material
   /// overwrite this method.
   virtual void updateStress(RankTwoTensor & strain_increment,
                             RankTwoTensor & inelastic_strain_increment,
-                            RankTwoTensor & stress_new);
+                            RankTwoTensor & stress_new) = 0;
 
+  /// Sets the value of the global variable _qp for inheriting classes
   void setQp(unsigned int qp);
 
+  /// Retained as an empty method to avoid a warning from Material.C in framework
+  /// This method is unused in all inheriting classes and should not be overwritten.
+  void resetQpProperties();
```

Comment:
I think you should override resetProperties as well and mark both with `final` if you really plan to not implement this function and you do not want children class to use it either.

In the header you can do the following and then you don't need anything in the .C file.
```c++
void resetProperties() final {}
void resetQpProperties() final {}
```